test/Models/CesiumSpec.ts

```diff
@@ -127,68 +127,75 @@ describeIfSupported("Cesium Model", function () {
-          const prim = cesium.scene.primitives.get(i);
-          return prim.allTilesLoaded && prim._url;
-        })
-      );
-
-    // Return names of all datasources in the scene
-    const dataSourceNames = () =>
-      range(cesium.dataSources.length).map(
-        (i) => cesium.dataSources.get(i).name
-      );
+    const container2 = document.createElement("div");
+    container2.id = "cesium-test-container";
+    document.body.appendChild(container2);
 
-    // Return urls of all imagery providers in the scene
-    const imageryProviderUrls = () =>
-      range(cesium.scene.imageryLayers.length)
-        .map(
-          (i) => (cesium.scene.imageryLayers.get(i).imageryProvider as any).url
-        )
-        .reverse();
+    const terriaViewer2 = new TerriaViewer(
+      terria,
+      computed(() => items)
+    );
 
-    await runLater(() => {});
+    const cesium2 = new Cesium(terriaViewer2, container2);
 
-    // Test that we have added the correct items
-    expect(dataSourceNames()).toEqual(["ds1", "ds2", "ds3"]);
-    expect(tilesetUrls()).toEqual(["prim1", "prim2", "prim3"]);
-    expect(imageryProviderUrls()).toEqual(["img1", "img2", "img3"]);
+    try {
+      // Return urls of all tilesets in the scene
+      const tilesetUrls = () =>
+        filterOutUndefined(
+          range(cesium2.scene.primitives.length).map((i) => {
+            const prim = cesium2.scene.primitives.get(i);
+            return prim.allTilesLoaded && prim._url;
+          })
+        );
 
-    runInAction(() => items.splice(0, 2));
-    await runLater(() => {});
+      // Return names of all datasources in the scene
+      const dataSourceNames = () =>
+        range(cesium2.dataSources.length).map(
+          (i) => cesium2.dataSources.get(i).name
+        );
 
-    // Test that we have removed the correct items
-    expect(dataSourceNames()).toEqual(["ds3"]);
-    expect(tilesetUrls()).toEqual(["prim3"]);
-    expect(imageryProviderUrls()).toEqual(["img3"]);
+      // Return urls of all imagery providers in the scene
+      const imageryProviderUrls = () =>
+        range(cesium2.scene.imageryLayers.length)
+          .map(
+            (i) =>
+              (cesium2.scene.imageryLayers.get(i).imageryProvider as any).url
+          )
+          .reverse();
+
+      await runLater(() => {});
+
+      // Test that we have added the correct items
+      expect(dataSourceNames()).toEqual(["ds1", "ds2", "ds3"]);
+      expect(tilesetUrls()).toEqual(["prim1", "prim2", "prim3"]);
+      expect(imageryProviderUrls()).toEqual(["img1", "img2", "img3"]);
+
+      runInAction(() => items.splice(0, 2));
+      await runLater(() => {});
+
+      // Test that we have removed the correct items
+      expect(dataSourceNames()).toEqual(["ds3"]);
+      expect(tilesetUrls()).toEqual(["prim3"]);
+      expect(imageryProviderUrls()).toEqual(["img3"]);
+    } finally {
+      cesium2.destroy();
+      document.body.removeChild(container2);
+    }
```

Comment:
Change variables names to not clash, and wrap in try, finally to ensure cesium & container is cleaned up